core/src/apps/bitcoin/authorize_coinjoin.py

```diff
@@ -0,0 +1,67 @@
+from micropython import const
+
+from trezor import ui
+from trezor.messages.AuthorizeCoinJoin import AuthorizeCoinJoin
+from trezor.messages.Success import Success
+from trezor.strings import format_amount
+from trezor.ui.text import Text
+
+from apps.base import set_authorization
+from apps.common.confirm import require_confirm, require_hold_to_confirm
+from apps.common.paths import validate_path
+
+from . import addresses
+from .authorization import FEE_PER_ANONYMITY_DECIMALS, CoinJoinAuthorization
+from .common import BIP32_WALLET_DEPTH
+from .keychain import get_keychain_for_coin
+
+if False:
+    from trezor import wire
+
+_MAX_COORDINATOR_LEN = const(18)
+
+
+async def authorize_coinjoin(ctx: wire.Context, msg: AuthorizeCoinJoin,) -> Success:
+    keychain, coin = await get_keychain_for_coin(ctx, msg.coin_name)
```

Comment:
https://github.com/trezor/trezor-firmware/pull/1127/commits/0a7721084fc0481ec49bc3558635f0d8f529455b Added comment and also realized I should wrap the whole thing in a `try` block to ensure the destruction of the keychain in case of an exception. Make sure to hide whitespace changes in the diff.